src/TaylorTrunc.jl

```diff
@@ -0,0 +1,279 @@
+export TaylorParam
+export taylorcircuit, get_param_type, taylorsolve, circuit_init, circuit_final, circuit_intermediate
+export v,lc
+
+const C(m, x, opn, t, c) = norm(x)*(opn*t*c)^(m)/factorial(m)
+const D(m, x, opn, t, c) = norm(x)*(opn*t*c)^(m-1)*t/factorial(m)
+struct TaylorParam{CPType, UType, L, HM, VType, S1Type, S2Type, WType}
+    k::Int # Taylor expansion upto k
+    t::L
+    H::HM
+    l::Int
+    rs::Int
+    C_tilda::CPType
+    D_tilda::CPType
+    N::CPType
+    V::VType
+    VS1::S1Type # Q is input size of GMblock
+    VS2::S2Type
+    VT::WType
+
+    function TaylorParam(k::Int, t::L, H::Matrix, x::Array{CPType,1}) where {L,CPType}
+        opn = opnorm(H)
+        u = isunitary(H/opn)
+        VT = nothing
+        C_tilda = 0
+        if u
+            for i in 0:k
+                C_tilda = C_tilda + C(i, x, opn,t,1)
+            end
+            N = C_tilda
+            C_tilda = sqrt(C_tilda)
+            rs = log2i(k+1)
+            VS1 = calc_vs1(k,rs,x,t,opn, C_tilda)
+            l = 0
+        else
+            for i in 0:k
+                C_tilda = C_tilda + C(i, x, opn,t,2)
+            end
+            N = C_tilda
+            C_tilda = sqrt(C_tilda)
+            rs = k
+            l = 2
+            VT = calc_vt(CPType)
+            VS1 = calc_vs1(k,x,t,opn, C_tilda)
+        end
+
+        new{CPType, u, L, Array{CPType,2}, Nothing, typeof(VS1), Nothing, typeof(VT)}(k, t, H, l, rs, C_tilda, zero(C_tilda), N, nothing, VS1, nothing,VT)
```

Comment:
If you initialize this with `Nothing` then what's the point of having this parameter?

Reply:
This is something that was bothering me as well. `Taylor Param` is either constructed with a matrix input or a `QuLDEProblem` input. In the former there is no need for `VS2`, `V`, or `N` but these parameters are required in the second case. I thought it was better to wrap it into `TaylorParam`, rather than have these parameters be called separately every time I need them in a function.